ddtrace/contrib/dbapi/__init__.py

```diff
@@ -36,10 +37,21 @@ class TracedCursor(wrapt.ObjectProxy):
     def __init__(self, cursor, pin, cfg):
         super(TracedCursor, self).__init__(cursor)
         pin.onto(self)
-        name = pin.app or "sql"
-        self._self_datadog_name = "{}.query".format(name)
         self._self_last_execute_operation = None
         self._self_config = cfg or config.dbapi2
+        # For most dbapi-compatible libraries, cursor method names can be
+        # prefixed with `config.<integration>._default_service` set by the
+        # integration for the library. However, in the case that an integration
+        # needs to distinguish these two, the
+        # `config.<integration>._method_name_prefix` is used.
```

Comment:
Not a huge fan of allowing multiple ways to do something. Would it be better to just force the integrations to explicitly define the prefix?

Reply:
Agree that multiple ways is less than desirable.

I'm reluctant to enforce this unless we know it's what we want to adopt across dbapi integrations. The use of `Pin.app` provided users a lot of flexibility in modifying the prefix for span names. And this only applied to the cursor methods and not connection methods, which used the vendor name which was not configurable.

I see two options:

a. add attribute to config (e.g., "config._method_name_prefix")
b. add property to `TracedCursor` that can be overridden by integrations (e.g., `TracedCursor.db_name`)

While (a) is more flexible (no need to subclass and just pass in a config object), I think (b) is preferable for maintenance and testing. Also, if this is in fact a value not intended to be changed, (b) would be preferred.